File: src/test/java/application/browser/HandHelper.java
package application.browser;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.matag.player.PlayerType;

public class HandHelper extends AbstractCardContainerHelper {
  private final PlayerType playerType;

  HandHelper(MatagBrowser matagBrowser, PlayerType playerType) {
    super(matagBrowser);
    this.playerType = playerType;
  }

  @Override
  protected WebElement containerElement() {
    if (playerType == PlayerType.PLAYER) {
      return matagBrowser.findElement(By.id("player-hand"));
    } else {
      return matagBrowser.findElement(By.id("opponent-hand"));
    }
  }
}

File: src/test/kotlin/application/browser/HandHelper.kt
package application.browser

import com.matag.player.PlayerType
import org.openqa.selenium.By
import org.openqa.selenium.WebElement

class HandHelper internal constructor(matagBrowser: MatagBrowser, private val playerType: PlayerType) :
    AbstractCardContainerHelper(matagBrowser) {
    override fun containerElement(): WebElement {
        return if (playerType == PlayerType.PLAYER) {
            matagBrowser.findElement(By.id("player-hand"))
        } else {
            matagBrowser.findElement(By.id("opponent-hand"))
        }
    }
}
